pallets/asset/src/lib.rs

```diff
@@ -1691,7 +1728,7 @@ impl<T: Trait> Module<T> {
             at > Self::total_checkpoints_of(&ticker)
         {
             // No checkpoints data exist
-            return Self::balance(&ticker, &did);
+            return Self::balance(&ticker, did).total;
```

Comment:
Call directly `Self::balance_of(ticker, &did)` because we are  not using `portfolio` value and we can save some time in the access to the storage.